apis/meta/reference_types.go

```diff
@@ -69,19 +69,74 @@ type SecretKeyReference struct {
 	Key string `json:"key,omitempty"`
 }
 
-// KubeConfigReference contains enough information to locate the referenced
-// Kubernetes secret that contains a kubeconfig file.
+const (
+	// KubeConfigKeyProvider is the key in the ConfigMap that contains the provider name.
+	KubeConfigKeyProvider = "provider"
+	// KubeConfigKeyAddress is the key in the ConfigMap that contains the cluster resource
+	// name in the provider API
+	KubeConfigKeyCluster = "cluster"
+	// KubeConfigKeyAddress is the key in the ConfigMap that contains the address of the
+	// Kubernetes API server.
+	KubeConfigKeyAddress = "address"
+	// KubeConfigKeyCACert is the key in the ConfigMap that contains the PEM-encoded CA
+	// certificate for the Kubernetes API server.
+	KubeConfigKeyCACert = "ca.crt"
+	// KubeConfigKeyAudiences is the key in the ConfigMap that contains the audiences
+	// for the Kubernetes ServiceAccount token.
+	KubeConfigKeyAudiences = "audiences"
```

Comment:
I think this should be a single line comma separated list to avoid any YAML formatting issue.

Reply:
Cloud providers have default audiences but they all allow customizing them. Generic clusters also have a default but also allow customizing.

Reply:
Using line separator will result in `aud1\n\raud2` in-cluster after KC applies it. It's way better to use a single line which will ensure readability in both Git and cluster.

Reply:
Worked for me:

```
k get cm testcm -o yaml
apiVersion: v1
data:
  audiences: |
    aud1
    aud2
kind: ConfigMap
metadata:
  creationTimestamp: "2025-07-14T09:18:56Z"
  labels:
    kustomize.toolkit.fluxcd.io/name: flux-system
    kustomize.toolkit.fluxcd.io/namespace: flux-system
  name: testcm
  namespace: flux-system
  resourceVersion: "228752"
  uid: 2accc570-e622-4521-ba50-afd7e53ab28d
```

Reply:
Hmm maybe kustomize build changed. Now going back to the initial ask, how many audiences would someone use to force them into this format?

Reply:
I suppose three URLs would already look too bad in one line, so 3? Also I'm not aware of comma restrictions in audiences, so we would lose the ability to represent them.

Reply:
> I'm not aware of comma restrictions in audiences, so we would lose the ability to represent them.

Oh Ok, let's go with line separator then.